### PR TITLE
chore: Restore self-deleting messages SQSERVICES-1059

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1059" title="SQSERVICES-1059" target="_blank">TaskSQSERVICES-1059</a>  Restore self-deleting messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Restore self-deleting messages
**FYI:** In this PR I revert only one my commit: [d65e354a2a8fdff3bf7d27de6072521bb5a985fc](https://github.com/wireapp/wire-ios/commit/d65e354a2a8fdff3bf7d27de6072521bb5a985fc), but as I can see there are not only for self-deleting messages changes here. Looks like I revert [this commit](https://github.com/wireapp/wire-ios/commit/8ce31a53f43a9cf486ed8677ed09f8c01cbe8993) in a previous PR. But I don't know why, because it wasn't the last commit. 
I double-checked the previous [PR](https://github.com/wireapp/wire-ios/pull/5401) and everything else is for self-deleting messages.

Needs releases with:

- [ wireapp/wire-ios-data-model/pull/1293 ] 
- [ wireapp/wire-ios-request-strategy/pull/432 ] 
- [ wireapp/wire-ios-share-engine/pull/299 ] 
- [ wireapp/wire-ios-sync-engine/pull/1586 ] 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 github "wireapp/wire-ios-ziphy" ~> 17.0.0
 github "wireapp/Cartography" "4.0.0_xcframework"
-github "wireapp/wire-ios-sync-engine" ~> 401.0.0
-github "wireapp/wire-ios-share-engine" ~> 236.0.0
+github "wireapp/wire-ios-sync-engine" ~> 400.0.0
```
Comment (typfel) [inline Cartfile:3]: You are degrading the SE?

**Review by typfel — APPROVED** [review]

File: Cartfile (hunk 1 above)
Comment (KaterinaWire) [reply Cartfile:3]: I will change Cartfile after all frameworks are released.

File: Wire-iOS Tests/UserConnectionViewTests.swift [hunk 2]
```diff
@@ -29,10 +29,12 @@ extension UIView {
 
 final class UserConnectionViewTests: XCTestCase {
 
-    func sutForUser(_ mockUser: MockUserType = SwiftMockLoader.mockUsers().first!) -> UserConnectionView {
+    func sutForUser(_ mockUser: MockUserType = SwiftMockLoader.mockUsers().first!, isFederated: Bool = false) -> UserConnectionView {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/UserConnectionViewTests.swift:32]: These seem unrelated?

File: Wire-iOS Tests/UserNameDetailViewTests.swift [hunk 3]
```diff
@@ -19,43 +19,49 @@
 import XCTest
 @testable import Wire
 
-class UserNameDetailViewTests: ZMSnapshotTestCase {
+class UserNameDetailViewTests: XCTestCase {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/UserNameDetailViewTests.swift:22]: This seems unrelated?

File: Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift [hunk 4]
```diff
@@ -712,18 +712,18 @@ final class StartConversationMatcher: TypedConversationStatusMatcher {
 }
 
 // Fallback for empty conversations: showing the handle.
-final class UnsernameMatcher: ConversationStatusMatcher {
+final class UsernameMatcher: ConversationStatusMatcher {
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift:715]: This seems unrelated?

File: Wire-iOS Tests/UserConnectionViewTests.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS Tests/UserConnectionViewTests.swift:32]: I explained this in the  PR description

File: Wire-iOS Tests/UserNameDetailViewTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS Tests/UserNameDetailViewTests.swift:22]: I explained this in the  PR description

File: Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift (hunk 4 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/ConversationList/ZMConversation+Status.swift:715]: I explained this in the  PR description

File: Wire-iOS Tests/UserConnectionViewTests.swift (hunk 2 above)
Comment (johnxnguyen) [reply Wire-iOS Tests/UserConnectionViewTests.swift:32]: Sorry I missed that.

**Review by johnxnguyen — APPROVED** [review]
